lib/ssl/src/tls_record_1_3.erl

```diff
@@ -218,92 +201,82 @@ decode_cipher_text(#ssl_tls{type = Type}, _) ->
-            PendingMaxEarlyDataSize =
-                pending_early_data_size(PendingMaxEarlyDataSize0, Data),
+        #ssl_tls{type=?APPLICATION_DATA, fragment=Data} ->
+            PendingMaxEarlyDataSize = pending_early_data_size(PendingMaxEarlyDataSize0, Data),
             if PendingMaxEarlyDataSize < 0 ->
                     %% Too much early data received, send alert unexpected_message
                     ?ALERT_REC(?FATAL, ?UNEXPECTED_MESSAGE,
                                {too_much_early_data,
                                 {max_early_data_threshold_exceeded,
                                  PendingMaxEarlyDataSize}});
                true ->
-                    ConnectionStates =
-                        ConnectionStates0#{current_read =>
-                                               ReadState0#{sequence_number => Seq + 1,
-                                                           pending_early_data_size => PendingMaxEarlyDataSize}},
+                    EarlyData = EarlyData0#{pending_early_data_size => PendingMaxEarlyDataSize},
+                    ReadState = ReadState0#{sequence_number => Seq + 1, early_data => EarlyData},
+                    ConnectionStates = ConnectionStates0#{current_read => ReadState},
                     {Record#ssl_tls{early_data = true}, ConnectionStates}
             end
     end.
 
-inner_plaintext(Type, Data, Length) ->
-    #inner_plaintext{
-       content = Data,
-       type = Type,
-       zeros = zero_padding(Length)
-      }.
-zero_padding(Length)->
-    binary:copy(<<?BYTE(0)>>, Length).
-
-encode_plain_text(#inner_plaintext{
-                     content = Data,
-                     type = Type,
-                     zeros = Zeros
-                    }, #{cipher_state := #cipher_state{key= Key,
-                                                       iv = IV,
-                                                       tag_len = TagLen},
-                         sequence_number := Seq,
-                         security_parameters :=
-                             #security_parameters{
-                                cipher_type = ?AEAD,
-                                bulk_cipher_algorithm = BulkCipherAlgo}
-                        }) ->
-    PlainText = [Data, Type, Zeros],
+encode_plain_text(Type, Data, 0,
+                  #{current_write :=
+                        #{cipher_state :=
+                              #cipher_state{key= Key,
+                                            iv = IV,
+                                            tag_len = TagLen},
+                          sequence_number := Seq,
+                          security_parameters :=
+                              #security_parameters{
+                                 cipher_type = ?AEAD,
+                                 bulk_cipher_algorithm = BulkCipherAlgo}
+                         } = Write} = CS) ->
+    %% Pad = <<0:(Length*8)>>,
+    PlainText = [Data, Type],  %% ++ Pad (currently always zero)
```

Comment:
You did not address prvious comment that was:

RFC refers to this [Data, Type, ZeroPad] as TLSInnerPlainText https://datatracker.ietf.org/doc/html/rfc8446 section 5.2
maybe we should call variable that to keep reference to RFC text?

Reply:
I never changed the name, fixed it now.